### ISSUE TITLE
[Bug] (CalDav) Shared calendar event invitations Accept/Decline not working

### ISSUE THREAD

**Issue by SebastianKrupinski** [nextcloud/server#48511]
### Steps to reproduce

1. Create shared calendar as user A and share with user B and C
2. Create event in shared calendar as User B and invite User C
3. Accept invitation from the email as User C

### Expected behavior

When clicking Accept/Decline the participant status in the event in the shared calendar and personal calendar should be updated.

### Actual behaviour

Currently the participant status is only updated in the personal calendar.

### Web server

None

### Database engine version

None

### PHP engine version

None

### Updated from an older installed version or fresh install

None

**Comment by SebastianKrupinski** [comment]
Test results.

![Screenshot 2024-10-01 174414](https://github.com/user-attachments/assets/b72f3898-9914-4e6d-9c73-c3b090f525ff)

![Screenshot 2024-10-01 174430](https://github.com/user-attachments/assets/d2c26b42-9649-439d-b9d7-ea17e1633ea9)

**Comment by SebastianKrupinski** [comment]
Issue is caused by the OCA\DAV\Controller\InvitationResponseController::buildITipResponse() method. The generated iTIp message is formatted incorrectly, as the sender and recipient are the same user.

![Screenshot 2024-10-01 180430](https://github.com/user-attachments/assets/85c143c7-f5b7-4c80-9daf-dfa3ff8f157b)

 The iTip message should be from sender "user1" to recipient "user3" (the organizer)

**Comment by jeekajoo** [comment]
I have exactly the same problem with your scenario :

* Create shared calendar as user A and share with user B and C
* Create event in shared calendar as User B and invite User C
* Accept invitation from the email as User C

When users are clicking on "Accept" to accept invitation : there is no error message. It just say it's successfully validated.
However the result is still  "Awaiting response" :

![Image](https://github.com/user-attachments/assets/a9d5e354-0783-4df6-ab6d-e3f9228e32bc)

Thank you for reporting this issue.

Nextcloud Hub 10, 31.0.4

**Comment by SebastianKrupinski** [comment]
> I just found out that if I change the organizer with the owner of the calendar : RSVP works if already invited people click on their invite link again.

Yes this would work, because it essentially changes the logic of who owns the event.
